src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -326,56 +326,92 @@ private static bool IsExpressionOfArgumentInDeconstruction(ExpressionSyntax expr
 
             expression = GetExpressionToAnalyzeForWrites(expression);
 
-            if (expression.IsOnlyWrittenTo())
-                return true;
-
-            if (expression.IsInRefContext(out var refParent))
+            while (true)
```

Comment:
the new analyzer helped identify some gaps in our existing helper.  Running on roslyn.sln produced several cases where we errantly looked just at the expr alone, and now what we might be doing on a value within it.